packages/design-system-react/src/components/button-base/ButtonBase.test.tsx

```diff
@@ -0,0 +1,97 @@
+    expect(screen.getByText('Button')).toHaveClass('h-10');
+
+    rerender(<ButtonBase size={ButtonBaseSize.Lg}>Button</ButtonBase>);
+    expect(screen.getByText('Button')).toHaveClass('h-12');
+  });
+
+  it('renders as child component when asChild is true', () => {
+    render(
+      <ButtonBase asChild>
+        <a href="#" target="_blank" rel="noopener noreferrer">
+          Link
+        </a>
+      </ButtonBase>,
+    );
+    const link = screen.getByRole('link');
+    expect(link).toBeInTheDocument();
+    expect(link).toHaveAttribute('href', '#');
+    expect(link).toHaveAttribute('target', '_blank');
+    expect(link).toHaveAttribute('rel', 'noopener noreferrer');
+  });
+
+  it('applies full width class when isFullWidth is true', () => {
+    render(<ButtonBase isFullWidth>Full Width Button</ButtonBase>);
+    expect(screen.getByText('Full Width Button')).toHaveClass('w-full');
+  });
+
+  it('shows loading state with loading text', () => {
+    render(
+      <ButtonBase isLoading loadingText="Please wait...">
+        Submit
+      </ButtonBase>,
+    );
+    expect(screen.getByText('⌛')).toBeInTheDocument();
+    expect(screen.getByText('Please wait...')).toBeInTheDocument();
+    expect(screen.queryByText('Submit')).not.toBeInTheDocument();
+  });
+
+  it('shows loading state with children when no loading text provided', () => {
+    render(<ButtonBase isLoading>Submit</ButtonBase>);
+    expect(screen.getByText('⌛')).toBeInTheDocument();
+    expect(screen.getByText('Submit')).toBeInTheDocument();
+  });
+
+  it('renders start icon when startIconName is provided', () => {
+    render(<ButtonBase startIconName="add-square-fill">With Icon</ButtonBase>);
+    const icon = screen.getByTestId('icon-add-square-fill');
+    expect(icon).toBeInTheDocument();
+    expect(icon).toHaveClass('mr-2');
+  });
+
+  it('renders end icon when endIconName is provided', () => {
+    render(<ButtonBase endIconName="arrow-right">With Icon</ButtonBase>);
+    const icon = screen.getByTestId('icon-arrow-right');
+    expect(icon).toBeInTheDocument();
+    expect(icon).toHaveClass('ml-2');
+  });
+
+  it('renders start accessory when provided', () => {
+    render(<ButtonBase startAccessory="→">With Accessory</ButtonBase>);
+    const accessory = screen.getByText('→');
+    expect(accessory).toBeInTheDocument();
+    expect(accessory).toHaveClass('mr-2');
+  });
+
+  it('renders end accessory when provided', () => {
+    render(<ButtonBase endAccessory="←">With Accessory</ButtonBase>);
+    const accessory = screen.getByText('←');
+    expect(accessory).toBeInTheDocument();
+    expect(accessory).toHaveClass('ml-2');
+  });
+
+  it('applies disabled state', () => {
+    render(<ButtonBase isDisabled>Disabled Button</ButtonBase>);
+    const button = screen.getByRole('button');
+    expect(button).toBeDisabled();
+    expect(button).toHaveClass('opacity-50', 'cursor-not-allowed');
+  });
+});
```

Comment:
Ensure 100% coverage of props

Reply:
<img width="1512" alt="Screenshot 2024-12-10 at 4 32 41 PM" src="https://github.com/user-attachments/assets/6aa051db-4225-4fa4-867a-97b1ec826b17">